sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -100,3 +100,15 @@
 // clang emits "warning:" in the message pragma output
 #define __SYCL_WARNING(msg) __pragma(message(msg))
 #endif // __GNUC__
+
+// defined __SYCL_UNROLL to add pragma/attribute unroll to a loop
+#ifndef __SYCL_UNROLL
+#if defined(__clang__) || defined(__INTEL_COMPILER)
+#define __SYCL_UNROLL(x) _Pragma(unroll(x))
+#elif (defined(__GNUC__) && __GNUC__ >= 8) ||                                  \
+    (defined(__GNUG__) && __GNUG__ >= 8)
+#define __SYCL_UNROLL(x) _Pragma(GCC unroll(x))
```

Comment:
The `_Pragma` operator takes a string literal (http://eel.is/c++draft/cpp.pragma.op#1), so I don't think this is quite correct. Above there is a `__SYCL_GCC_PRAGMA` macro, that strikes me as what you want to use here.

Reply:
Yeah, also there should be a stringification to process 'x' properly. The last patch worked in the compiler explorer at least for all the compilers I claim to have unroll pragma.